src/components/draw/DrawDirective.js

```diff
@@ -161,165 +317,228 @@ goog.require('ga_map_service');
 
             // Remove interactions
             deactivateDrawInteraction();
-            deactivateSelectInteraction();
-            deactivateModifyInteraction();
+            select.setActive(false);
+
+            // Unregister the events attached to the source
+            for (var i in unSourceEvents) {
+              ol.Observable.unByKey(unSourceEvents[i]);
+            }
+            unSourceEvents = [];
+
           };
 
-          // Deactivate other tools
-          var activateTool = function(tool) {
-            layer.visible = true;
+          // Set the draw interaction with the good geometry
+          var deregDrawStart, deregDrawEnd, deregPointerMove2;
+          var activateDrawInteraction = function(tool) {
```

Comment:
The problem is the draw interaction is different between all tools and because of the help tooltip and all the evnts to register deregister,  we need some global variables on all event changefeature, drawstart, drawend. 

I think we could but if you're ok I prefer to do it after deploy to avoid breaking something.